src/components/Header/Header.tsx

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import * as todoSevice from '../../api/todos';
+import { Errors } from '../../types/Errors';
+import { handleRequestError } from '../../utils/handleRequestError';
+import { useTodosContext } from '../../utils/useTodosContext';
+
+export const Header: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    setTempTodo,
+    setError,
+    setLoadingTodoIds,
+    activeTodos,
+    completedTodos,
+    isLoading,
+    setIsloading,
+  } = useTodosContext();
+  const addTodoInputRef = useRef<HTMLInputElement>(null);
+  const isClassActive = completedTodos.length > 0 && activeTodos.length === 0;
+  const [title, setTitle] = useState('');
+  const [isSubMit, setIsSubmit] = useState(false);
+
+  useEffect(() => {
+    if (isLoading && addTodoInputRef.current) {
+      addTodoInputRef.current.focus();
+      setIsloading(false);
+    }
+  }, [setIsloading, isLoading]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setError(Errors.default);
+    setTitle(event.target.value);
+  };
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+    setIsSubmit(true);
+
+    if (!title.trim().length) {
+      handleRequestError(Errors.emptyTitle, setError);
+      setIsSubmit(false);
+    } else {
+      const newTodo = {
+        title: title.trim(),
+        userId: todoSevice.USER_ID,
+        completed: false,
+      };
+
+      const tempTodo = {
+        id: 0,
+        ...newTodo,
+      };
+
+      setTempTodo(tempTodo);
+      setLoadingTodoIds([tempTodo.id]);
+      todoSevice
+        .createTodo(newTodo)
+        .then((response: Todo) => {
+          setTodos((prevTodos: Todo[]) => [...prevTodos, response]);
+          setTempTodo(null);
+          setLoadingTodoIds([]);
+          setTitle('');
+          setIsloading(true);
+        })
+        .catch(() => {
+          handleRequestError(Errors.addTodo, setError);
+          setTempTodo(null);
+          setLoadingTodoIds([]);
+          setIsloading(true);
```

Comment:
The same as I mentioned previously:
- use functional `setState` to set/remove loading todo ids
- The operations that you do it in both cases do in the finally block